Diagnostic/diagnostic.py

```diff
@@ -750,7 +758,17 @@ def install_omi():
 
     if need_install_omi:
         hutil.log("Begin omi installation.")
-        RunGetOutput(distConfig["installomi"])
+        isOmiInstalledSuccessfully = False
+        for trialNum in range(1, 4):   # Retry up to 3 times to install OMI
+            isOmiInstalledSuccessfully = RunGetOutput(distConfig["installomi"])[0] is 0
+            if isOmiInstalledSuccessfully:
+                break;
+            hutil.error("OMI install failed (trial #" + str(trialNum) + "). Retrying in 30 seconds...")
+            time.sleep(30)
+        if not isOmiInstalledSuccessfully:
+            hutil.error("OMI install failed 3 times. Giving up...")
```

Comment:
@jasonzio : range(1,4) in python is [1,2,3], so I believe it's going to try 3 times...?